user-guide/Advanced_Modules/User Defined APIs/UD_APIs_Troubleshooting.md

```diff
@@ -0,0 +1,89 @@
+  - `UserDefinableApiEndpointInstaller.txt` contains logs from the installation of the DxM if it is installed by a DataMiner upgrade package or via the command line. This will not contain additional logs when it is installed with its GUI. A small subset of these log lines will also be available in the `progress.log` DataMiner upgrade package log file and will be displayed during the upgrade.
+- `C:\Skyline DataMiner\Logging\SLUserDefinableApiManager.txt` contains logging from the manager in SLNet that manages the CRUD actions on API tokens/definitions and the API triggers.
+
+> [!TIP]
+> The API triggers arriving on an endpoint are forwarded randomly to one of the DataMiner agents for load balancing when there are more than 2 agents in the DMS. Keep this in mind when checking the logging. If an HTTP request was sent to the endpoint running on a specific server, there is no guarantee that this trigger will be handled by the DataMiner agent on that same server. The JSON error responses returned by the endpoint should include a 'faultingNode' field which hints at the ID of the DMA that actually handled the request.
+
+Below you can find the request flow when a trigger comes in. This may help you during your troubleshooting.
+
+```mermaid
+flowchart LR
+client("User") -- "HTTP(S) request" --> a
+subgraph "IIS"
+    a("IIS") --> b("Rewrite Rule")
+end
+subgraph "Extension"
+    b -- HTTP request --> c("UserDefinableApiEndpoint\nDxM")
+end
+subgraph "Core DataMiner"
+    c -- NATS --> d("UserDefinableApiEndpointManager\n(SLNet.exe)") 
+    d -- IPC --> e("SLAutomation.exe")
+end
+```
+
+## Installation issues
+
+The errors in this section can be generated when the UserDefinableApiEndpoint DxM is installed via a DataMiner upgrade package. These will be visible in the upgrade log during the upgrade and also in the `UserDefinableApiEndpoint.txt` log file as mentioned in the [General](#general) section.
+
+### 'Could not find AspNetCore installed on this system'
+
+DataMiner should install ASP.NET Core 5 automatically, but in case this failed or it has been removed, the installer could give this error. To fix this, go to the [download page](https://dotnet.microsoft.com/en-us/download/dotnet/5.0) of .NET 5 and download the latest ASP.NET Core Runtime 5. After that, you can manually rerun the installer which can be found in this  location: `C:\Skyline DataMiner\Tools\ModuleInstallers\DataMiner UserDefinableApiEndpoint 1.X.X.X.msi`.
+
+### 'Found rewrite rule, but has inconsistencies:'
+
+The log line(s) after it will contain the inconsistencies of the IIS rewrite rule. This rewrite rule forwards requests coming in on port 80 or 443 to the UserDefinableApiEndpoint DxM. It should contain the correct port where the DxM is currently running on. The installer will always verify if the port mentioned in the rewrite rule matches the one that is configured in the [app settings of the DxM](xref:UD_APIs_UserDefinableApiEndpoint#configuring-the-dxm).
```

Comment:
```suggestion
The log line(s) after it will contain the inconsistencies of the IIS rewrite rule. This rewrite rule forwards requests coming in on port 80 or 443 to the UserDefinableApiEndpoint DxM. It should contain the correct port where the DxM is currently running on. The installer will always verify if the port mentioned in the rewrite rule matches the one that is configured in the [appsettings file of the DxM](xref:UD_APIs_UserDefinableApiEndpoint#configuring-the-dxm).
```